components/eam/src/physics/cam/clubb_intr.F90

```diff
@@ -1492,6 +1499,12 @@ subroutine clubb_tend_cam( &
 #endif
    det_s(:)   = 0.0_r8
    det_ice(:) = 0.0_r8
+
+   if (macmic_it > 99) then
+      call endrun('clubb_tend_cam: macmic_it > 99. Revise checkpoint name for cnd_diag_checkpoint.')
+   end if
+   write(char_macmic_it,'(i2.2)') macmic_it
```

Comment:
Alternatively, if to extend CondiDiag for potential use  independent of CLUBB, then moving the declaration of char_macmic_it outside #ifdef CLUBB_SGS would sound better. Either way is good for fixing the current issue.

Reply:
For MMF and IDEAL configurations it would be nice if we could avoid building any CLUBB code (and any code not needed by these packages), but I recall looking into this when refactoring the MMF physpkg and deciding that it was tricky. However, that was before the change to cmake, so it might be easier now.

Reply:
I see what the issue was: I did not pay attention to `#ifdef CLUBB_SGS` in `clubb_intr.F90`.

I'd be interested in getting educated (probably offline) on the reasons for having `#ifdef CLUBB_SGS` inside `clubb_intr.F90`. Perhaps the old build system was the reason, as (@whannah1 mentioned)?